agent/agent.cpp

```diff
@@ -203,19 +204,34 @@ Agent::Agent(
 }
 
 
-Device *Agent::findDeviceByUUIDorName(const std::string &aId)
+const Device * Agent::getDeviceByName(const std::string& name) const
```

Comment:
I added these two methods to allow correct std::map lookup behaviour within this class. Existing usages of m_deviceMap[{name}] within this class should now use these methods.